tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/jdbc/JdbcTableManager.java

```diff
@@ -73,18 +79,79 @@ public TableInstance createImmutable(JdbcTableDefinition tableDefinition, TableH
     {
         TableName tableName = createImmutableTableName(tableHandle);
         LOGGER.debug("creating immutable table {}", tableName);
-        dropTableIgnoreError(tableName); // ultimately we do not want to drop table here - if data did not change
-        createTable(tableDefinition, tableName);
-        JdbcTableDataSource dataSource = tableDefinition.getDataSource();
-        insertData(tableName, dataSource);
+        // TODO: drop and recreate table if the underlying data has changed
+        if (!tableExists(tableName)) {
+            createTable(tableDefinition, tableName);
+            JdbcTableDataSource dataSource = tableDefinition.getDataSource();
+            insertData(tableName, dataSource);
+        }
         return new JdbcTableInstance(tableName, tableDefinition);
     }
 
+    private boolean tableExists(TableName tableName)
+    {
+        Connection connection = queryExecutor.getConnection();
+        try {
+            DatabaseMetaData metadata = connection.getMetaData();
+            String escape = metadata.getSearchStringEscape();
+            try (ResultSet resultSet = metadata.getTables(
+                    connection.getCatalog(),
+                    escapeNamePattern(tableName.getSchema().orElse(null), escape),
+                    escapeNamePattern(tableName.getName(), escape),
+                    new String[] {"TABLE"})
+            ) {
+                while (resultSet.next()) {
+                    if (tableNameMatches(tableName, resultSet) && tableSchemaMatches(tableName, resultSet)) {
+                        LOGGER.info("Table {} already exists", tableName.getNameInDatabase());
```

Comment:
move this `LOGGER.info` statement to `createImmutable` and add `" skipping creating immutable table"` to log message.